### PR TITLE
feat(reasoner): support remove duplicate

### PR DESCRIPTION
**Pull request by wangsff**

### CODE REVIEW

**Review by youdonghai — APPROVED** [review]
LGTM

File: reasoner/lube-logical/src/main/scala/com/antgroup/openspg/reasoner/lube/logical/operators/Select.scala [hunk 1]
```diff
@@ -15,7 +15,11 @@ package com.antgroup.openspg.reasoner.lube.logical.operators
 
 import com.antgroup.openspg.reasoner.lube.logical.{SolvedModel, Var}
 
-final case class Select(in: LogicalOperator, fields: List[Var], as: List[String])
+final case class Select(
+    in: LogicalOperator,
+    fields: List[Var],
+    as: List[String],
+    isDistinctGet: Boolean)
```
Comment (fishjoy) [inline reasoner/lube-logical/src/main/scala/com/antgroup/openspg/reasoner/lube/logical/operators/Select.scala:22]: isDistinctGet -> distinct

File: reasoner/lube-physical/src/main/scala/com/antgroup/openspg/reasoner/lube/physical/operators/Select.scala [hunk 2]
```diff
@@ -21,8 +21,9 @@ import com.antgroup.openspg.reasoner.lube.physical.rdg.RDG
 final case class Select[T <: RDG[T]: TypeTag](
     in: PhysicalOperator[T],
     orderedFields: List[Var],
-    as: List[String])
+    as: List[String],
+    isDistinctGet: Boolean)
```
Comment (fishjoy) [inline reasoner/lube-physical/src/main/scala/com/antgroup/openspg/reasoner/lube/physical/operators/Select.scala:25]: isDistinctGet -> distinct

File: reasoner/lube-physical/src/main/scala/com/antgroup/openspg/reasoner/lube/physical/operators/Select.scala [hunk 3]
```diff
@@ -21,8 +21,9 @@ import com.antgroup.openspg.reasoner.lube.physical.rdg.RDG
...
     extends PhysicalOperator[T] {
-  def row: T#Records = in.rdg.select(orderedFields, as)
+  def row: T#Records = in.rdg.select(orderedFields, as, isDistinctGet)
```
Comment (fishjoy) [inline reasoner/lube-physical/src/main/scala/com/antgroup/openspg/reasoner/lube/physical/operators/Select.scala:27]: there is no In RDG, it is not necessary to add "distinct" to the select interface; "distinct" should be placed within the row interface.

Comment (fishjoy) [reply reasoner/lube-physical/src/main/scala/com/antgroup/openspg/reasoner/lube/physical/operators/Select.scala:27]: eg, in.rdg.select(orderedFields, as).distinct()

**Review by fishjoy — APPROVED** [review]
LGTM

**Pull request merged**